### PR TITLE
Sort users and orgs on explore by recency by default

### PR DESCRIPTION
**Pull request by techknowlogick**
This gives more "freshness" to the explore page. So it's not just the same X users on the explore page by default, now it matches the same sort as the repos on the explore page.

### CODE REVIEW

**Comment by wxiaoguang** [comment]
IIRC the code is reused in many places, does it change the admin user list page behavior?

**Comment by techknowlogick** [comment]
@wxiaoguang This specific code is only reused in explore: https://sourcegraph.com/search?q=context%3Aglobal+repo%3A%5Egithub%5C.com%2Fgo-gitea%2Fgitea%24+RenderUserSearch&patternType=standard&sm=1&groupBy=path

**Comment by wxiaoguang** [comment]
image

image

**Comment by techknowlogick** [comment]
@wxiaoguang aha, you're right. whoops. I'll blame my bad vision on that. Although now thinking about it, the admin page sorts repos by recency as well, so this would match that behaviour too.

**Comment by wxiaoguang** [comment]
Could we make the admin user list still use `alphabetically` by default? Suppose I am a site admin and is browsing users, I can filter users page by page by `alphabetically` and review every username. But by `recentupdate`, the names seems a little disorganized.

**Comment by techknowlogick** [comment]
@wxiaoguang I've made that change, but I'm unsure about the changes I made, I tried to do it in the most "DRY" way, but there might be a better way. Thoughts?

**Comment by wxiaoguang** [comment]
Maybe:

```
const UserSearchDefaultExploreSort = "recentupdate"
const UserSearchDefaultAdminSort = "alphabetically"
```

Then LGTM

File: routers/web/admin/users.go [hunk 1]
```diff
@@ -53,14 +53,14 @@ func Users(ctx *context.Context) {
 
 	sortType := ctx.FormString("sort")
 	if sortType == "" {
-		sortType = explore.UserSearchDefaultSortType
+		sortType = "alphabetically"
```
Comment (wxiaoguang) [inline routers/web/admin/users.go]:
```suggestion
		sortType = explore.UserSearchDefaultAdminSort
```

**Review by wxiaoguang — APPROVED** [review]
The previous LGTM was done by comment, so approve again.

**Review by delvh — APPROVED** [review]

**Comment by lunny** [comment]
Maybe we should move the extra parameter as one field of the options?

**Comment by silverwind** [comment]
Unresolved comment above.

**Comment by techknowlogick** [comment]
@silverwind which comment?

**Comment by wxiaoguang** [comment]
@techknowlogick @lunny how to continue?

**Comment by lunny** [comment]
I will send some commits to this PR.

**Comment by lunny** [comment]
@techknowlogick @wxiaoguang @delvh please review again.

**Review by delvh — APPROVED** [review]

File: modules/context/form.go [hunk 2]
```diff
@@ -65,3 +65,8 @@ func (ctx *Context) FormOptionalBool(key string) util.OptionalBool {
 	v = v || strings.EqualFold(s, "on")
 	return util.OptionalBoolOf(v)
 }
+
+func (ctx *Context) SetFormValue(key, value string) {
```
Comment (wxiaoguang) [inline modules/context/form.go]: To match `FormString`, it should be `SetFormString`

And `Form.Set` is also `Set(key, value string)`, string

Comment (lunny) [reply modules/context/form.go]: done.

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**